client/server/pages/index.js

```diff
@@ -331,10 +331,12 @@ function setUpLoggedInRoute( req, res, next ) {
 
 				if ( data.localeSlug ) {
 					req.context.lang = data.localeSlug;
-					req.context.store.dispatch( {
-						type: LOCALE_SET,
-						localeSlug: data.localeSlug,
-						localeVariant: data.localeVariant,
+					import( 'calypso/state/ui/init' ).then( () => {
```

Comment:
`ui` is probably already loaded at this point, but I'm adding this here to make sure.

Reply:
Let's keep a close eye on this when testing once it lands in staging.

Reply:
Dropped this change to the server. It was breaking the server unit tests, and it shouldn't be necessary.

Reply:
Agreed